L1Trigger/L1TGlobal/src/GlobalBoard.cc

```diff
@@ -1016,10 +1018,9 @@ void l1t::GlobalBoard::runFDL(edm::Event& iEvent,
         // Make sure algo bit in range, warn otherwise
         if (iBit < prescaleFactorsAlgoTrig.size()) {
           if (prescaleFactorsAlgoTrig.at(iBit) != 1) {
```

Comment:
I did some investigation, and currently there is no dedicated error handling, although it also doesn't crash when a value like -3 or 0.65 is set as the prescale. Both cases never result in a final L1A  - the decision is always `false`. 

The reasons for both are different. For fractional values with 0 < value < 1, these lines [1] result in the trigger bit being masked (i.e. prescale set to 0), since the prescale value is cast to an int.  For negative values, the `prescale_count` in PrescaleCounter is a negative number cast to size_t, which wraps around and becomes an incredibly high prescale number. So with a sufficiently large number events there would eventually be a L1A.

I think it is a good idea to add a check, but I personally am not sure where is the best place to add one. Certainly it could be added to zeroPrescaleCounters and semirandomNumber. Alternately it could be added here [2], where the prescale is applied. It could also be added earlier, when the prescale table is being read in. I would lean towards [2]. 

[1] https://github.com/cms-sw/cmssw/blob/5222b47952a15160dd438f421170741529cc363e/L1Trigger/L1TGlobal/plugins/L1TGlobalProducer.cc#L578-#L582
[2] https://github.com/cms-sw/cmssw/blob/265f2b80ccf66df2d487b0536b44b278a57f309c/L1Trigger/L1TGlobal/src/GlobalBoard.cc#L1020